nixops/deployment.py

```diff
@@ -613,6 +616,14 @@ def get_physical_spec(self) -> Any:
             m.name: set() for m in active_machines.values()
         }
 
+        for plugin in get_plugins():
+            deployment_hooks = plugin.deployment_hooks()
+            if not deployment_hooks:
+                continue
+
+            for name, attrs in deployment_hooks.physical_spec(self).items():
+                attrs_per_resource[name].extend(attrs)
+
```

Comment:
I think we should not pepper too much of the plugin specific code here, and instead have something like:

```
plugin_manager.physical_spec(self)
```

to handle the iterating, skipping, etc. I'm not sure how to handle the extension of the attrs_per_resource. If we pass it in, it'll get modified, right?
```